types/backbone/index.d.ts

```diff
@@ -568,26 +568,28 @@ declare namespace Backbone {
 
         model: TModel;
         collection: Collection<TModel>;
-        // template: (json, options?) => string;
-        setElement(element: HTMLElement | JQuery, delegate?: boolean): View<TModel>;
-        setElement(element: any): View<TModel>;
-        id: string;
+        setElement(element: HTMLElement | JQuery): this;
+        id?: string;
         cid: string;
-        className: string;
+        className?: string;
         tagName: string;
 
-        el: any;
+        el: HTMLElement;
         $el: JQuery;
-        attributes: any;
+        attributes: { [id: string]: any };
         $(selector: string): JQuery;
-        render(): View<TModel>;
-        remove(): View<TModel>;
-        delegateEvents(events?: EventsHash): any;
-        delegate(eventName: string, selector: string, listener: Function): View<TModel>;
-        undelegateEvents(): any;
-        undelegate(eventName: string, selector?: string, listener?: Function): View<TModel>;
-
-        _ensureElement(): void;
+        render(): this;
+        remove(): this;
+        delegateEvents(events?: EventsHash | (() => EventsHash)): this;
+        delegate(eventName: string, selector: string, listener: Function): this;
+        undelegateEvents(): this;
+        undelegate(eventName: string, selector?: string, listener?: Function): this;
+
+        private _removeElement(): void;
+        private _setElement(el: HTMLElement | JQuery): void;
+        private _createElement(tagName: string): void;
+        private _ensureElement(): void;
+        private _setAttributes(attributes: { [id: string]: any }): void;
```

Comment:
While it intuitively makes sense, it isn't technically correct to declare these methods private. Although these methods are internal, they are publicly accessible, and some libraries (such as backbone-relational and marionette) rely on their presence. If the developers of those libraries were using TypeScript, a change like this could be a massive pain for them. Regardless, it would be a breaking change, which I suspect is also difficult (if not impossible) to work around. So please remove the `private` keyword here.

Reply:
I'm suggest to make them `protected`, if this is really required.

Reply:
Well if you really insist on sticking a qualifier on these methods, then I guess that `protected` won't hurt much. I think `protected` and `private` basically *never* apply, though, unless the library in question was written with TypeScript or some fancy Babel frontend which actually has such semantics. Plain ES6 is just public-only. But I guess I'm rowing against the stream as far as this is concerned.